apollo-runtime/src/main/java/com/apollographql/apollo/ApolloClient.java

```diff
@@ -151,6 +154,20 @@ public ApolloStore apolloStore() {
     return apolloStore;
   }
 
+  /**
+   * Sets the idleCallback which will be called when this ApolloClient is idle.
+   */
+  public void setIdleCallback(Runnable idleCallback) {
```

Comment:
I feel it shouldn't be a part of the builder because it will force the clients to supply the callback at the time of creation of the apolloClient which I feel is a bit limiting. For example if it is being used in conjunction with a DI framework, the clients might not have the access to callback at all, for example in a Dagger2 module which contains a @Provides method for creating ApolloClient.